test/integration/stripe_checkout_test.rb

```diff
@@ -0,0 +1,34 @@
+require "test_helper"
+
+class StripeCheckoutTest < ActionDispatch::IntegrationTest
+  include PayHelper
+
+  setup do
+    @user = users(:with_business)
+    sign_in @user
+  end
+
+  test "redirects to the Stripe Checkout URL" do
+    stub_pay(@user) do
+      post stripe_checkout_path
+      assert_redirected_to "checkout.stripe.com"
+    end
+  end
+
+  test "passes along the plan" do
+    full_time_price_id = Rails.application.credentials.stripe[:price_ids][:full_time_plan]
+    stub_pay(@user, plan_price_id: full_time_price_id) do
+      post stripe_checkout_path(plan: :full_time)
+    end
+  end
+
+  test "redirects to the stored location on success" do
+    success_path = new_developer_message_path(Developer.first)
+    get success_path
+
+    stub_pay(@user) do
+      post stripe_checkout_path
+      assert_equal Analytics::Event.last.url, success_path
+    end
+  end
```

Comment:
This test is failing and I can't figure out why. The `stub_pay` (from `test/support/helpers/pay_helper.rb`) isn't picking up on the `user` stub. I _think_ because we stub `@user` but the controller fetches via `current_user` so they aren't the same instance (although they are the same record).

I also can't set `session` directly until a request is made. Which might be a red herring now that I type it out.